sql/ts/src/skdb_util.ts

```diff
@@ -102,3 +104,42 @@ export class SKDBTable extends Array<Record<string, any>> {
     });
   }
 }
+
+/* ***************************************************************************/
+/* Transaction API for programmatically constructing and executing
+   transactions against an SKDB instance */
+/* ***************************************************************************/
+
+export class SKDBTransaction {
+  private query_params: Params;
+  private stmts: Array<string>;
+  private db_handle: SKDB;
+
+  constructor(skdb: SKDB) {
+    this.query_params = {};
+    this.stmts = [];
+    this.db_handle = skdb;
+  }
+
+  addStatement(stmt: string): SKDBTransaction {
```

Comment:
Could we call this 'add'? Just because you might want to add queries or statements, and it's shorter to type. :D

Reply:
Sounds good to me on the shorter-to-type front!  Though I'd quibble that queries are (syntactically) just statements, not some separate thing :)